modules/4337/test/erc4337/Safe4337Module.spec.ts

```diff
@@ -82,22 +87,63 @@ describe('Safe4337Module', () => {
       }
       const operationHash = await validator.getOperationHash(userOp)
 
+      const gasParams = (changes: Partial<GasParameters>) => packGasParameters({ ...unpackGasParameters(userOp), ...changes })
       for (const [name, value] of [
         ['sender', '0x0000000000000000000000000000000000000021'],
         ['nonce', 0x22],
         ['initCode', '0x23'],
         ['callData', '0x24'],
-        ['accountGasLimits', packAccountGasLimits(0x26, 0x25)],
-        ['preVerificationGas', 0x27],
-        ['maxFeePerGas', 0x28],
-        ['maxPriorityFeePerGas', 0x29],
+        ['preVerificationGas', 0x25],
+        ['accountGasLimits', gasParams({ verificationGasLimit: 0x26 }).accountGasLimits],
+        ['accountGasLimits', gasParams({ callGasLimit: 0x27 }).accountGasLimits],
+        ['gasFees', gasParams({ maxPriorityFeePerGas: 0x28 }).gasFees],
+        ['gasFees', gasParams({ maxFeePerGas: 0x29 }).gasFees],
         ['paymasterAndData', '0x2a'],
         ['signature', signature({ validAfter: 0x2b, validUntil: 0x1c })],
         ['signature', signature({ validAfter: 0x1b, validUntil: 0x2c })],
-      ]) {
+      ] as const) {
         expect(await validator.getOperationHash({ ...userOp, [name]: value })).to.not.equal(operationHash)
       }
     })
+
+    it('should match locally computed Safe operation hash', async () => {
```

Comment:
how is this different from the test on line 41 in the same file?

Reply:
Oh, I missed that existing test. It should do the same thing. The issue is that the existing test uses the default Safe operation values, which in particular means `maxFeePerGas == maxPriorityFeePerGas` and does not (as per the PR description):

> [...] catch issues with field orders being slightly different in the `EncodedSafeOp` struct that we would otherwise miss. Notably, the "fee per gas" parameters are always set to the same value, and our existing test suite was not able to catch swapped order in `EncodedSafeOp` fields.

I'm going to remove this new test case and replace the existing test with this new logic.